test/aries-js-worker/test/didexchange/edgeagent_router.js

```diff
@@ -31,54 +97,201 @@ describe("DID-Exchange between an Edge Agent and a router", function() {
         aries.startNotifier(notice => {
             try {
+                assert.isObject(notice)
+                assert.property(notice, "isErr")
                 assert.isFalse(notice.isErr)
                 assert.property(notice, "payload")
-                const connection = notice.payload
-                assert.property(connection, "connection_id")
-                aries.didexchange.acceptInvitation({
-                    id : connection.connection_id
-                })
+                assert.property(notice.payload, "state")
             } catch (err) {
                 done(err)
             }
-            done()
+            if (notice.payload.state === "completed") {
+                done()
+            }
         }, ["connections"])
-        aries.didexchange.receiveInvitation(invitation)
     })
+})
 
-    it("Edge Agent validates that the connection's state is 'completed'", function(done) {
-        aries.startNotifier(notice => {
+describe("DID-Exchange between two Edge Agents using the router", function () {
+    var aliceAgent, bobAgent
+    var invitation, connectionID
+
+    before(async () => {
+        return new Promise((resolve, reject) => {
+            newAries('alice').then(
+                a => {
+                    aliceAgent = a;
+                    resolve()
+                },
+                err => reject(new Error(err.message))
+            )
+
+            newAries('bob').then(
+                a => {
+                    bobAgent = a;
+                    resolve()
+                },
+                err => reject(new Error(err.message))
+            )
+        })
+    })
+
+    after(() => {
+        aliceAgent.destroy()
+        bobAgent.destroy()
+    })
+
+    it(`Router is running on "${routerHttpUrl},${routerWsUrl}" with controller "${routerControllerApiUrl}"`, async function () {
+        await routerHealthCheck(routerHttpUrl, routerWsUrl, routerControllerApiUrl)
+    })
+
+    it("Alice Edge Agent receives an invitation from the router via the controller API", async function () {
+        const response = await axios.post(routerCreateInvitationPath)
+        invitation = response.data.invitation
+
+        validateInvitation(invitation)
+    })
+
+    it("Alice Edge Agent accepts the invitation from the router", function (done) {
+        acceptInvitation(aliceAgent, invitation, done)
+    })
+
+    it("Alice Edge Agent validates that the connection's state is 'completed'", function (done) {
+        aliceAgent.startNotifier(notice => {
             try {
                 assert.isObject(notice)
                 assert.property(notice, "isErr")
                 assert.isFalse(notice.isErr)
                 assert.property(notice, "payload")
                 assert.property(notice.payload, "state")
+                assert.property(notice.payload, "state")
             } catch (err) {
                 done(err)
             }
             if (notice.payload.state === "completed") {
```

Comment:
since it looks like some extra refactoring was included in this PR, I would think the new constant declared at the top would be used for comparison here

Reply:
[Updated]([Fixed](https://github.com/hyperledger/aries-framework-go/compare/85fc76cec463acdf2a2d50767969fef8cd8eb093..5e233a431cda218ef4f3b53699aa1bf5abcd89b9).